fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -1422,7 +1425,7 @@ private void computeMultiCastFragmentParams() throws Exception {
 
                 List<TPlanFragmentDestination> destinations = multiSink.getDestinations().get(i);
                 if (enablePipelineEngine && enableShareHashTableForBroadcastJoin
-                        && params.fragment.isRightChildOfBroadcastHashJoin()) {
+                        && destFragment.isRightChildOfBroadcastHashJoin()) {
```

Comment:
according to the silimar code in `computeFragmentExecParams`, this code block is to want to decide how to shuffle data to dest by whether it is broadcast join's right child. Since `MultiCastFragment` will never be broadcast join's right child. I think the right way to fix this problem is removing if branch.

Reply:
be supports cte as bc right.